extension/app/ts/background/settings.ts

```diff
@@ -159,3 +162,28 @@ export async function savePendingInterceptorAccessRequestPromise(promise: Pendin
 	return await browser.storage.local.set({ InterceptorAccessRequestPromise: PendingInterceptorAccessRequestPromise.serialize(promise) })
 }
 
+const SIMULATION_RESULTS_STORAGE_KEY = 'simulationResults'
+export async function getSimulationResults() {
+	const results = await browserStorageLocalGet(['simulationResults'])
+	if (results.simulationResults === undefined) {
+		return {
+			simulationId: 0,
+			simulationState: undefined,
+			visualizerResults: undefined,
+			addressBookEntries: [],
+			tokenPrices: [],
+			activeAddress: undefined
+		}
+	}
+	return SimulationResults.parse(results.simulationResults)
+}
+
+const simulationResultsSemaphore = new Semaphore(1)
+export async function updateSimulationResults(newResults: SimulationResults) {
+	simulationResultsSemaphore.execute(async () => {
+		const oldResults = await getSimulationResults()
+		// TODO: use funtypes to parse/validate `results` here.  `browser.storage.local.get` returns an `any` I think (which is terrible)
```

Comment:
Looks like this was forgotten, see other comment about the actual types allowed though.
```suggestion
```